### ISSUE TITLE
Request: Automatically set desired line endings when saving file

### ISSUE THREAD

**Issue by hhallen** [geany/geany#984]
As far as I can find, Geany will only use the line endings specified in "Preferences / Files / Default end of line characters" when it creates new files. Existing files will keep their line endings, unless you check the "Ensure consistent line endings" box, in which case I suppose Geany will find out what type of line ending is in the majority, and use that throughout the file.
What I would like is an option to always, unconditionally, force the default line ending to be applied when saving any file, whether newly created or previously existing.

**Comment by codebrainz** [comment]
It's not automatic or a preference, but I believe `Document->Set Line Endings->Convert and Set to <kind>` will go through the document and convert each line-ending to `<kind>`, making them all the same.

**Comment by elextr** [comment]
IIRC from a conversation a loooong while ago *automagically* changing line
ends was judged to be unexpected some of the time, and so it was left as a
manual operation as Matthew noted.

**Comment by hhallen** [comment]
Sure, you can do it manually. But I don't see how this is much different from, for example, automagically trimming trailing whitespace. And I don't think you can argue that it would be "unexpected" if you have to tick a checkbox in Preferences to enable it.

**Comment by jonlandrum** [comment]
I agree with @hhallen on this one. It's not unexpected if it's an opt-in setting.

**Comment by elextr** [comment]
A well written opt-in pull request may be accepted, or at least will cause the issue to be re-considered.

**Comment by jonlandrum** [comment]
@elextr thank you, I'll dig into the code over the weekend and see what it would take.
